tests/Doctrine/Migrations/Tests/Generator/DiffGeneratorTest.php

```diff
@@ -118,6 +119,60 @@ static function ($name) {
         self::assertSame('path', $this->migrationDiffGenerator->generate('1234', '/table_name1/', true, 80));
     }
 
+    public function testGenerateFromEmptySchema() : void
+    {
+        $schemaConfig = $this->createMock(SchemaConfig::class);
+        $toSchema     = $this->createMock(Schema::class);
+        $table        = $this->createMock(Table::class);
+
+        $this->schemaManager->expects(self::once())
+            ->method('createSchemaConfig')
+            ->willReturn($schemaConfig);
+
+        $schemaConfig->method('getName')->willReturn('schema');
+
+        $this->schemaManager->expects(self::never())
+            ->method('createSchema');
+
+        $this->schemaProvider->expects(self::once())
+            ->method('createSchema')
+            ->willReturn($toSchema);
+
+        $this->dbalConfiguration->expects(self::once())
+            ->method('getSchemaAssetsFilter')
+            ->willReturn(null);
+
+        $toSchema->method('getName')->willReturn('schema');
+        $toSchema->method('getTables')->willReturn(['schema.table_name' => $table]);
+        $toSchema->method('getTable')->with('table_name')->willReturn($table);
+        $toSchema->method('getNamespaces')->willReturn([]);
+        $toSchema->method('getSequences')->willReturn([]);
+
+        $table->method('getName')->willReturn('schema.table_name');
+        $table->method('getShortestName')->with('schema')->willReturn('table_name');
```

Comment:
After thinking, that part of the test is too fragile (tightly coupled to internal DBAL implementation). I'll rework the implementation (during the weekend)